Domain/Transaction.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization.Formatters;
+
+using Types;
+
+namespace Domain
+{
+    public abstract class Transaction
+    {
+#pragma warning disable 8618
+        protected Transaction() { } // Make EFCore happy
+#pragma warning restore 8618
+        protected Transaction(
+            string counterPartyName, 
+            Guid bankAccountId, 
+            decimal amount, 
+            DateTimeOffset receivedDateTime, 
+            string description, 
+            int sequence, 
+            ICollection<TransactionAttachment> attachments, 
+            Guid? memberId,
+            ICollection<TransactionTypeAmount> transactionTypeAmounts)
+        {
+            Id = Guid.NewGuid();
```

Comment:
Couldn't the `sequence` property act as the primary key for this class?

Reply:
Yes, but if I'm not mistaken, the sequence can be dropped, since we can just order by creation date.